packages/flame_3d/lib/src/extensions/vector2.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:flame_3d/game.dart';
+
+/// Represents an immutable [Vector2].
+typedef ImmutableVector2 = ({double x, double y});
```

Comment:
A bit confusing to call this `ImmutableVector2` since it's not a `Vector2` object

Reply:
No it is it's own type 😉, it is still a vector value type imho. Just not one that has many methods (on purpose)